A few things

jamesplease opened the pull request:
- Swaps in `rimraf`. Presently, `npm run build` will error out if you do not already have a `dist` directory. Most new contributors will not have a `list` directory, which makes it hard for them to build the lib to see what the result is! There are ways to do this by writing a fancier shell script, but rimraf is nice because it supports many different operating systems.

  I am happy to replace this with another solution if you'd prefer. Let me know!

- Adds lock files to the gitignore. Self-explanatory, I guess. I can remove this commit if you prefer.

- Removes prop-types from production builds. This reduces the file size from `792 B` to `728 B`, which is roughly a 10% file size reduction.

  I didn't look too closely at the build step to determine if there are situations when you might want the prop types to remain in the production build.

Thanks for reviewing! :v:

jamesplease committed:
Use rimraf instead of rm for deleting the dist files

jamesplease committed:
Add lock files to gitignore

jamesplease committed:
Remove prop-types for production build

jamesplease commented on package.json:
```diff
@@ -29,6 +29,7 @@
   },
   "devDependencies": {
     "babel-plugin-external-helpers": "6.22.0",
+    "babel-plugin-transform-react-remove-prop-types": "^0.4.13",
```
I can lock down these version numbers if you would like. Also, we could add an `.npmrc` with `save-exact=true` so that this preference would apply to all future contributors. What do you think?

erikthedeveloper replied:
I do prefer exact versions. I think I set that on my CLI config a while back but didn't know about the `.npmrc` option. That is good to know. Locking these versions down and adding the `.npmrc` would be a great addition here!

erikthedeveloper reviewed (commented):
Very good stuff @jamesplease! I've been meaning to revisit some of this so thanks for the nudge and PR. I'll pull this down tomorrow and check it out further along with your file size reduction idea. Thank you 😃

erikthedeveloper commented on .gitignore:
```diff
@@ -1,2 +1,5 @@
 node_modules
 dist
+package-lock.json
+package-lock.json.*
```
I don't see any harm in explicitly ignoring npm-specific files, but I wonder if it is necessary since the project is yarn-based so I would avoid merging these in in any case 

🤔 Related tangent: I've preferred yarn, but I have wondered if I should stick just to npm for libs/open source stuff

jamesplease replied:
> but I wonder if it is necessary since the project is yarn-based so I would avoid merging these in in any case

I think that it is necessary, since as far as I know it is impossible to actually enforce yarn for other contributors. You may be using it locally, but a contributor using npm will generate a package lock file.

i.e. I typically just default to npm when contributing to open source projects, so I pulled this project down, ran `npm i`, and a package-lock file was generated for me.

> Related tangent: I've preferred yarn, but I have wondered if I should stick just to npm for libs/open source stuff

I think that you can use either for your own development (it really doesn't matter what you use, I think), but I think it is cool when the project is set up to support a developer using either.

erikthedeveloper replied:
Yeah, good point. I will keep that in mind for future. If I'm going to favor yarn in my projects then I think I need to ignore `package-lock*`, track `yarn.lock`, and make sure `yarn.lock` gets updated anytime `package.json` dependencies are changed (I can update this myself for contributors not using `yarn`). I do like tracking a lock file but would want to avoid tracking/maintaining both.

I think I'll continue with that for now, but I I'm going to let the question of just sticking with npm stew further if it might simplify things/contributions.

erikthedeveloper commented on .gitignore:
```diff
@@ -1,2 +1,5 @@
 dist
+package-lock.json
+package-lock.json.*
+yarn.lock
```
I typically track `yarn.lock` (it is already checked in) since it ensures consistent installs across machines so I think we need to drop this line. From https://yarnpkg.com/lang/en/docs/yarn-lock/#toc-check-into-source-control

> All yarn.lock files should be checked into source control (e.g. git or mercurial). This allows Yarn to install the same exact dependency tree across all machines, whether it be your coworker’s laptop or a CI server.
>
>Framework and library authors should also check yarn.lock into source control. Don’t worry about publishing the yarn.lock file as it won’t have any effect on users of the library.

jamesplease replied:
> I typically track yarn.lock (it is already checked in)

Ah, cool, cool. I should have checked before I added this file to the gitignore. I will remove this from the gitignore :v:

erikthedeveloper commented on package.json:
```diff
@@ -11,7 +11,7 @@
     "prebuild": "npm run clean",
     "build": "NODE_ENV=production rollup -c",
-    "clean": "rm dist/*",
+    "clean": "rimraf dist/*",
```
Very good! I'll keep `rimraf` in mind for next time I publish anything 👍

erikthedeveloper commented on .babelrc:
```diff
@@ -5,7 +5,13 @@
+        "external-helpers",
+        [
+          "transform-react-remove-prop-types",
+          { "mode": "remove", "removeImport": true }
```
We are definitely safe to "remove" here. No use cases for needing to keep the PropTypes around 👍

erikthedeveloper replied:
@jamesplease one more thought to consider 🤔 If we "remove" propTypes from the build altogether (including CJS, ESM, UMD) then we lose the benefit of having propTypes at all since users will no longer receive warnings when providing invalid props, even when they are running in development mode. It appears to me that this is removing propTypes for all `dist/*.js` output files.

I'm pretty new to `remove-prop-types` myself. Would it be beneficial to _remove_ for the `.umd.min.js` build to reduce file size and _wrap_ for other builds to enable propTypes warnings? When consumers are building their apps for production, I'm assuming their build step will strip out any dead blocks such as `if (NODE_ENV !== 'production') {/* ... */}` which leaves it open for libs such as `react`, `react-render-watch`, etc... to publish propTypes behind the flag for removal.

I'm curious what you think about this.

---

Thinking about this further, I wonder if this is the fault of my setup/`.babelrc` since the "production" settings are seemingly being used for all builds 😳 haha actually I think that might be it...

erikthedeveloper reviewed (changes requested):
Thanks again for the PR and input @jamesplease! I left some comments throughout. I pulled it down and it built as expected. 

Only 2 concerns: I would rather keep tracking `yarn.lock` and stick to exact versions. 

I'm happy to make these edits myself and push onto this branch if you don't have the time, but I'll hold off a bit so we don't cross any wires.

jamesplease commented:
Thank you for the review, @erikthedeveloper ! I should have some time to update this PR in the next few days.

Also, very cool library!

erikthedeveloper committed:
Remove yarn.lock from .gitignore

erikthedeveloper committed:
Add .npmrc with save-exact=true

erikthedeveloper committed:
Use exact versions in package.json

erikthedeveloper committed:
Update yarn.lock

erikthedeveloper commented:
@jamesplease I feel like we are on the same page here on changes to be made so I'll take care of those and push them up now. I'll wait for any further thoughts/feedback from you before merging.

erikthedeveloper committed:
yarn add cross-env

erikthedeveloper committed:
"remove" propTypes for umd.min.js otherwise "wrap"

jamesplease committed:
Merge pull request #1 from erikthedeveloper/build-stuffs

Revisit build setup and remove vs. wrap propTypes

erikthedeveloper committed:
.babelrc nest all rules under env entries

This fixes the tests being broken due to "babelHelpers" being undefined.
It seems like the top level rules were not being merged into the per-env
rules as expected.

erikthedeveloper commented:
Somehow the tests are failing to run for me locally and it seems due to how I have the `.babelrc` set up.

The really weird thing is that I swear I already ran into this, poked around to find a fix, and implemented it as part of https://github.com/jamesplease/react-render-watch/pull/1. I must have just imagined it or forgot to commit somewhere haha. I'll push that change onto this branch now and get this merged in.

**Before**

![image](https://user-images.githubusercontent.com/1240178/37034966-3bc2a698-2108-11e8-826f-ba9c5aff053d.png)

**After**

![image](https://user-images.githubusercontent.com/1240178/37035211-05659fe6-2109-11e8-9e80-a60a9996b4e2.png)

erikthedeveloper commented:
Released as part of [v0.3.0](https://github.com/erikthedeveloper/react-render-watch/releases/tag/v0.3.0) thanks again @jamesplease!
